### PR TITLE
Extend LLM connection to support other engines

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

Extend LLM connection to support other engines

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dvilelaf — APPROVED** [review]

File: packages/valory/connections/openai/connection.py [hunk 1]
```diff
@@ -166,21 +171,36 @@ def _get_response(self, prompt_template: str, prompt_values: Dict[str, str]):
...
+            response = openai.ChatCompletion.create(
+                model=engine,
+                messages=messages,
+                temperature=self.openai_settings["temperature"],
+                max_tokens=self.openai_settings["max_tokens"],
+                n=1,
+                timeout=120,
+                stop=None,
+            )
+            output = response.choices[0].message.content
```
Comment (Adamantios) [inline packages/valory/connections/openai/connection.py:191]: No `timeout` for a single hanging request is given.

File: packages/valory/connections/openai/connection.py [hunk 2]
```diff
@@ -166,21 +171,36 @@ def _get_response(self, prompt_template: str, prompt_values: Dict[str, str]):
...
+        else:
```
Comment (Adamantios) [inline packages/valory/connections/openai/connection.py:192]: Is `else` enough? What if the given value does not exist?

File: packages/valory/connections/openai/connection.py (hunk 1 above)
Comment (jmoreira-valory) [reply packages/valory/connections/openai/connection.py:191]: Can you elaborate on that?

Comment (Adamantios) [reply packages/valory/connections/openai/connection.py:191]: You need to set the [request_timeout kwarg](https://github.com/openai/openai-python/blob/e7ee4daaa5b2f4b90c519faee993377ab691559d/openai/api_resources/abstract/engine_api_resource.py#L160) for that. The [timeout](https://github.com/valory-xyz/mech/blob/main/tools/openai_request.py#L64) that you are currently using only takes effect [when retrying](https://github.com/openai/openai-python/blob/e7ee4daaa5b2f4b90c519faee993377ab691559d/openai/api_resources/completion.py#L26-L28).

Comment (Adamantios) [reply packages/valory/connections/openai/connection.py:191]: It has already been done.

File: packages/valory/connections/openai/connection.yaml [hunk 3]
```diff
@@ -19,6 +19,7 @@ config:
   max_tokens: 500
   openai_api_key: null
   temperature: 0.7
+  request_timeout: 300
```
Comment (Adamantios) [inline packages/valory/connections/openai/connection.yaml:22]: Please override `request_timeout` at the agent and the service as well.

Comment (jmoreira-valory) [reply packages/valory/connections/openai/connection.yaml:22]: Fixed

**Review by Adamantios — APPROVED** [review]

**Comment by dvilelaf** [comment]
Ported to: https://github.com/valory-xyz/IEKit/pull/97

**Comment by dvilelaf** [comment]
Ported to https://github.com/valory-xyz/IEKit/pull/97

**Pull request closed without merging**